src/future/mod.rs

```diff
@@ -46,15 +46,20 @@
 //! [`Future::race`]: trait.Future.html#method.race
 //! [`Future::try_race`]: trait.Future.html#method.try_race
 
-pub use future::Future;
-pub use pending::pending;
-pub use poll_fn::poll_fn;
-pub use ready::ready;
+cfg_alloc! {
+    pub use future::Future;
+    pub(crate) mod future;
+}
+
+cfg_std! {
+    pub use pending::pending;
+    pub use poll_fn::poll_fn;
+    pub use ready::ready;
```

Comment:
FYI: to enable these utilities on no-std, use manually implemented futures instead of async functions.